persistence-jdbc/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/persistence/jdbc/JdbcPersistenceComponents.scala

```diff
@@ -44,6 +44,7 @@ private[lagom] trait SlickProviderComponents extends DBComponents {
     )(executionContext)
     new SlickProvider(actorSystem, coordinatedShutdown)(executionContext)
   }
+  slickProvider
```

Comment:
why don't you simply remove the `lazy` keyword?

Reply:
Because I'm eager. Still, this Pr may be incomplete as the eagerness/laziness of `slickProvider` may only be a [red herring](https://github.com/lagom/lagom/issues/2947#issuecomment-658134091).

Reply:
I understand that you want it to be eagerly evaluated. My point is, remove the `lazy` and you don't have to eagerly call `slickProvider` at the end. And we don't pay the `lazy val` tax.

Reply:
😂 I got your point. I was joking to try to divert the attention from the fact that my code-edit was dumb.

Reply:
@renatocaval 
Is removing the `lazy` binary compatible? As long as I remember, I tried and got the error from MiMa.

Reply:
> Is removing the lazy binary compatible? As long as I remember, I tried and got the error from MiMa. 

I would be surprised. Code access `slickProvider` is not aware that it's `lazy` and that's exactly the feature.

Reply:
Ah! I now remember I only used this approach because I say the original commit by @ihostage and just rushed through the change. 

Bincompat is a good reason. I was curious and just cheked. `lazy val` becomes 

```
  public static com.lightbend.lagom.internal.persistence.jdbc.SlickProvider slickProvider$(com.lightbend.lagom.scaladsl.persistence.jdbc.SlickProviderComponents);
  public com.lightbend.lagom.internal.persistence.jdbc.SlickProvider slickProvider();
```

while `val` is:

```
  public abstract com.lightbend.lagom.internal.persistence.jdbc.SlickProvider slickProvider();
```

and that is exactly what the mima report is highlightling:


```
[error] lagom-scaladsl-persistence-jdbc: Failed binary compatibility check against com.lightbend.lagom:lagom-scaladsl-persistence-jdbc_2.12:1.6.1! Found 2 potential problems
[error]  * abstract method slickProvider()com.lightbend.lagom.internal.persistence.jdbc.SlickProvider in interface com.lightbend.lagom.scaladsl.persistence.jdbc.SlickProviderComponents does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectAbstractMethodProblem]("com.lightbend.lagom.scaladsl.persistence.jdbc.SlickProviderComponents.slickProvider")
[error]  * abstract synthetic method com$lightbend$lagom$scaladsl$persistence$jdbc$SlickProviderComponents$_setter_$slickProvider_=(com.lightbend.lagom.internal.persistence.jdbc.SlickProvider)Unit in interface com.lightbend.lagom.scaladsl.persistence.jdbc.SlickProviderComponents is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("com.lightbend.lagom.scaladsl.persistence.jdbc.SlickProviderComponents.com$lightbend$lagom$scaladsl$persistence$jdbc$SlickProviderComponents$_setter_$slickProvider_=")
[error] stack trace is suppressed; run last persistence-jdbc-scaladsl / mimaReportBinaryIssues for the full output
```

Reply:
I think it's safe to add the exclusion. From user perspective, that's just a field accessible through `slickProvider` to be injected whenever needed. Users will recompile their code anyway. 

Only an issue if someone builds a library on wrapping SlickComponent. I don't think we should make it more complicated then needed.

Reply:
See my comment below. I've rolled back this change as it's not the solution.